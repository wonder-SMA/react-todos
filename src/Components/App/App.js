import React, { useState, useContext } from 'react';
import styled from 'styled-components';

import { StoreContext } from '../../index';
import Input from '../Input';
import Header from '../Header';
import TodoHeader from '../TodoHeader';
import TodoList from '../TodoList';
import TodoToolBar from '../TodoToolBar';

const AppWrapper = styled.div`
  width: 100%;
  max-height: 100vh;
  height: 100vh;
  padding: 28px;
  background-color: #f5f5f5;

  svg {
    cursor: pointer;
  }

  *:not(:first-child, :last-child) {
    p {
      margin-left: 14px;
      font-size: 26px;
    }
  }

  @media (min-width: 576px) {
    *:not(:first-child, :last-child) {
      p {
        margin-left: 18px;
        font-size: 30px;
      }
    }
  }

  @media (min-width: 740px) {
    width: 740px;
    margin: 0 auto;
  }
`;

function App() {
  const { store } = useContext(StoreContext);
  const [isClosed, setIsClosed] = useState(false);

  const addTodoHandler = (todo) => {
    if (todo) {
      store.setAllTodos(todo);
      store.setFilterValue('All');
    }
  };

  const closeListHandler = () => {
    setIsClosed(!isClosed);
  };

  return (
    <AppWrapper>
      <Header />
      <Input addTodoHandler={addTodoHandler} />
      <TodoHeader isClosed={isClosed} closeListHandler={closeListHandler} />
      <TodoList isClosed={isClosed} />
      <TodoToolBar />
    </AppWrapper>
  );
}

export default App;