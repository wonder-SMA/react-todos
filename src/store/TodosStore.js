import { makeAutoObservable } from 'mobx';

class TodosStore {
  _filterValue = 'All';
  _allTodos = [];
  _completedTodos = [];

  constructor() {
    makeAutoObservable(this);
  }

  get todos() {
    switch (this.filterValue) {
      case 'All':
        return this.allTodos;
      case 'Active':
        return this.activeTodos;
      case 'Completed':
        return this.completedTodos;
    }
  }

  get filterValue() {
    return this._filterValue;
  }

  setFilterValue(value) {
    this._filterValue = value;
  }

  get allTodos() {
    return this._allTodos;
  }

  setAllTodos(todo) {
    if (!this._allTodos.includes(todo)) {
      this._allTodos = [...this._allTodos, todo];
    }
  }

  get activeTodos() {
    return this._allTodos.filter(todo => !this._completedTodos.includes(todo));
  }

  get completedTodos() {
    return this._completedTodos;
  }

  setCompletedTodos(todo) {
    if (this.completedTodos.includes(todo)) {
      this._completedTodos = this.completedTodos.filter(completedTodo => !(completedTodo === todo));
    } else {
      this._completedTodos = [...this._completedTodos, todo];
    }
  }

  clearCompletedTodos() {
    this._allTodos = [...this.activeTodos];
    this._completedTodos = [];
  }
}

export default TodosStore;