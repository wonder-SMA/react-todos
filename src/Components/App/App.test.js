import React from 'react';
import { render } from '@testing-library/react';

import { StoreContext } from '../../index';
import TodosStore from '../../store';
import App from './';

describe('App', () => {
  test('App snapshot didn\'t change', () => {
    const component = render(
      <StoreContext.Provider value={{
        store: new TodosStore()
      }}>
        <App />
      </StoreContext.Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
