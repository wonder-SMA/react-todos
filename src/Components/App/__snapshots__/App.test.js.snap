// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App snapshot didn't change 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iBkjds kKDXLN"
      >
        <div
          class="sc-dkzDqf kkJGTB"
        >
          <h1>
            todos
          </h1>
        </div>
        <div
          class="sc-gsnTZi ixMWhz"
        >
          <input
            placeholder="Введите задачу"
            type="text"
            value=""
          />
          <button
            class="sc-bczRLJ fjmREE"
          >
            Add todo
          </button>
        </div>
        <div
          class="sc-hKMtZM ePirV"
        >
          <svg
            role="presentation"
            style="width: 2.25rem; height: 2.25rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
              style="fill: #545454;"
            />
          </svg>
          <p>
            What needs to be done?
          </p>
        </div>
        <ul
          class="sc-jSMfEi bxLorI"
        />
        <div
          class="sc-gKXOVf iTkNqA"
        >
          <p>
            0
             items left
          </p>
          <div>
            <button
              class="sc-bczRLJ fNXOZK"
            >
              All
            </button>
            <button
              class="sc-bczRLJ fjmREE"
            >
              Active
            </button>
            <button
              class="sc-bczRLJ fjmREE"
            >
              Completed
            </button>
            <button
              class="sc-bczRLJ ejnYZA"
            >
              Clear completed
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iBkjds kKDXLN"
    >
      <div
        class="sc-dkzDqf kkJGTB"
      >
        <h1>
          todos
        </h1>
      </div>
      <div
        class="sc-gsnTZi ixMWhz"
      >
        <input
          placeholder="Введите задачу"
          type="text"
          value=""
        />
        <button
          class="sc-bczRLJ fjmREE"
        >
          Add todo
        </button>
      </div>
      <div
        class="sc-hKMtZM ePirV"
      >
        <svg
          role="presentation"
          style="width: 2.25rem; height: 2.25rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
            style="fill: #545454;"
          />
        </svg>
        <p>
          What needs to be done?
        </p>
      </div>
      <ul
        class="sc-jSMfEi bxLorI"
      />
      <div
        class="sc-gKXOVf iTkNqA"
      >
        <p>
          0
           items left
        </p>
        <div>
          <button
            class="sc-bczRLJ fNXOZK"
          >
            All
          </button>
          <button
            class="sc-bczRLJ fjmREE"
          >
            Active
          </button>
          <button
            class="sc-bczRLJ fjmREE"
          >
            Completed
          </button>
          <button
            class="sc-bczRLJ ejnYZA"
          >
            Clear completed
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
